### DISCUSSION TITLE
Access parameter defined in metadata with python knitr block

### DISCUSSION THREAD

**Discussion by DamonCharlesRoberts** [quarto-dev/quarto-cli#4688] [Q&A] [answered]
I see that there are ways to access the parameter list that you may define in the metadata while using knitr, but I don't see the way I'd do this if I am using python.

Say I have a yaml section with the following

```
---
title: Title of project
params:
    windows: true
    reproduce: false
---
```

And I want to access that `params` list but with python. How would I do that? Is it a dictionary? A list?

**Comment by cderv** [comment, marked as answer]
For using parameters with Jupyter document, the syntax is different and params are not defined in YAML 
See https://quarto.org/docs/computations/parameters.html#jupyter

If by "python knitr block", you mean a `python` chunk in a Knitr document (which some R chunks), then you need to use **reticulate** feature [to access R from Python](https://rstudio.github.io/reticulate/articles/r_markdown.html#calling-r-from-python)

````markdown
---
title: Title of project
params:
  windows: true
  reproduce: false
---

```{r}
params
```

```{python}
r.params
```
````

Hope it helps

**Reply by DamonCharlesRoberts** [reply]
Okay. Gotcha. I'll probably just go with a python block using `knitr` then! Thanks for the example for extracting it with `reticulate`!
